### PR TITLE
Icds reports clean ui

### PR DESCRIPTION
**Pull request by dmydlo**
Hi @emord,
in this PR I have made changes to UI code in ICDS Reports so that we will have less redundant code.

### CODE REVIEW

**Comment by emord** [comment]
Wow @dmydlo this looks like a lot of effort and the +/- ratio is great. Thanks a bunch for doing this. 

This is a really large change, so it may take a while to review. Is there anyway that some of these changes could be broken apart into smaller change sets, so that they are easier to review?

**Comment by dmydlo** [comment]
Well, I think that breaking it into smaller changes could be done only in one way:
- adding new service (baseControllers)
for each of BaseController's methods or variables:
    - create method or variable in BaseContorller, delete the redundant piece of code from every one of 22 specific controllers, if needed fix tests
That would mean it could be represented by let's say 10 commits.

Would that be helpful?

**Comment by emord** [comment]
Yes. I think that would make this PR easier to review

File: custom/icds_reports/static/js/directives/medicine-kit/medicine-kit.directive.js [hunk 1]
```diff
@@ -1,228 +1,80 @@
...
-
-            yAxis: {
-                axisLabel: '',
-                tickFormat: function(d){
-                    return d3.format(".2%")(d);
-                },
-                axisLabelDistance: 20,
-                forceY: [0],
+    vm.chartOptions = vm.getChartOptions(
+        '%b %Y', ".2%", ' Percentage of AWCs that reported having a Medicine Kit'
```
Comment (lwyszomi) [inline custom/icds_reports/static/js/directives/medicine-kit/medicine-kit.directive.js]: I think this function should take as a parameter the object, something like:
```
vm.getChartOptions({
       xAxisFormat: '%b %Y',
       yAxisFormat: '.2%',
       title: 'Percentage of AWCs that reported having a Medicine Kit' 
})
```
Will be more readable.

**Comment by dmydlo** [comment]
I am closing this PR due to creating new one with step-by-step commits for easier review.
New [PR](https://github.com/dimagi/commcare-hq/pull/20088)

**Pull request closed without merging**